File: src/ast/DtorName.java
package ast;

import java.io.StringWriter;

public class DtorName extends ParentNode {
    public DtorName(BaseNode name) {
        super(NodeType.DtOrName, name);
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("~");
        child.PrintLeft(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/DtorName.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class DtorName(name: BaseNode?) : ParentNode(NodeType.DtOrName, name) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("~")
        child!!.printLeft(writer)
    }
}